File: pythia-0.8/examples/tabulator/tabulator/Quadratic.py
#!/usr/bin/env python
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#
#                             Michael A.G. Aivazis
#                      California Institute of Technology
#                      (C) 1998-2005  All Rights Reserved
#
# <LicenseText>
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#


from pyre.components.Component import Component


class Quadratic(Component):


    class Inventory(Component.Inventory):
        
        import pyre.inventory

        a = pyre.inventory.float("a", default=0.0)
        b = pyre.inventory.float("b", default=0.0)
        c = pyre.inventory.float("c", default=0.0)


    def initialize(self):
        self.a = self.inventory.a
        self.b = self.inventory.b
        self.c = self.inventory.c

        import tabulator._tabulator
        tabulator._tabulator.quadraticSet(self.a, self.b, self.c)
        return


    def __init__(self):
        Component.__init__(self, "quadratic", "functor")
        self.a = 0.0
        self.b = 0.0
        self.c = 0.0

        import tabulator._tabulator
        self.handle = tabulator._tabulator.quadratic()
        
        return


    def _init(self):
        Component._init(self)
        self.initialize()
        return


# version
__id__ = "$Id: Quadratic.py,v 1.1.1.1 2005/03/17 20:03:02 aivazis Exp $"

# End of file 

File: pythia-0.8/examples/tabulator/libtabulator/quadratic.f90
! -*- F90 -*-
!
! ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
!
!                             Michael A.G. Aivazis
!                      California Institute of Technology
!                      (C) 1998-2005  All Rights Reserved
!
! <LicenseText>
!
! ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
!

module quadratic_parameters
    double precision:: a, b, c
end module quadratic_parameters

!
subroutine quadratic_set(new_a, new_b, new_c)
    use quadratic_parameters

    implicit none
    double precision:: new_a, new_b, new_c

    a = new_a
    b = new_b
    c = new_c

    return
end subroutine quadratic_set

!
function quadratic(x)
    use quadratic_parameters

    implicit none
    double precision:: quadratic, x

    quadratic = a*x**2 + b*x + c

end function quadratic


! version
! $Id: quadratic.f90,v 1.1.1.1 2005/03/17 20:03:02 aivazis Exp $

! End of file 
